src/modules/Writers.jl

```diff
@@ -59,13 +59,91 @@ function render(io::IO, mime::MIME"text/plain", node::Expanders.DocsNodes, page,
+"""
+function join_decl(decl)
+    n, t = decl
+    isempty(t) ? n : "$(n)::$(t)"
+end
+
 function render(io::IO, mime::MIME"text/plain", node::Expanders.DocsNode, page, doc)
     # Docstring header based on the name of the binding and it's category.
     anchor = "<a id='$(node.anchor.id)' href='#$(node.anchor.id)'>#</a>"
     header = "**`$(node.object.binding)`** &mdash; *$(Utilities.doccat(node.object))*."
     println(io, anchor, "\n", header, "\n\n")
     # Body. May contain several concatenated docstrings.
     renderdoc(io, mime, node.docstr, page, doc)
+
+    # Table of methods.
+    # If DocsNode.methods is nulled, then we assume that we should not render a
+    # table. However, if the list of methods is there but has no elements then
+    # we output an appropriate note saying that the name has no methods associated
+    # with it.
+    Utilities.unwrap(node.methods) do ms
+        println(io, "<p><strong>Methods</strong></p>")
+
+        if isempty(ms)
+            println(io, "<p><emph>This function has no methods.</emph></p>")
+        end
+
+        print(io, """
+        <ul class="documenter-methodtable">
+        """)
+
+        # Determine the name of the method.
+        # Most generally the name is Module.Submodule.object, but we do not need
+        # to specify the modules here (anymore). Hence we just pick the name of
+        # the field, store in object.binding.var as a Symbol.
+        name = node.object.binding.var
+
+        # A regexp to match filenames with an absolute path
+        r = Regex("$(Pkg.dir())/([A-Za-z0-9]+)/(.*)")
+
+        for m in ms
+            tv, decls, file, line = Base.arg_decl_parts(m)
+            decls = decls[2:end]
+            file = string(file)
+            url = get(Utilities.url(doc.internal.remote, m.module, file, line), "")
+            file_match = match(r, file)
+            if file_match !== nothing
+                file = file_match.captures[2]
+            end
+            # We'll generate the HTML now. If the list of arguments is too long
+            # (which can happen quite easily due to the long type names), we will
+            # display them in a multiline block instead.
+            tvars = isempty(tv) ? "" : "{$(join(tv, ", "))}"
+            args = join([join_decl(d) for d in decls], ", ")
+            if length(args) <= 50
+                print(io, """
+                <li>
+                    <code><span class="n f">$(name)$(tvars)</span>($(args))</code>
+                    defined at
+                    <a href="$(url)">$(file):$(line)</a>
+                </li>
+                """)
+            else
+                args = join([(" "^4)*join_decl(d) for d in decls], ",\n")
+                print(io, """
+                <li>
+                    <pre><code><span class="n f">$(name)$(tvars)</span>(
+                $(args)
+                )</code></pre>
+                    defined at
+                    <a href="$(url)">$(file):$(line)</a>
```

Comment:
I find it useful to have these kind of URLs open in separate tabs, rather than the same one. Just needs a `target='_blank'` like `source` links for docstrings do.